### PR TITLE
fix: avoid consuming kibana response twice

### LINKED ISSUE #920: TypeError: unusable when creating a new project
Hello,

I'm unable to create a new project. When running `npx @elastic/synthetics init projects-test`, after entering the Kibana URL and the API key I get the error `TypeError: unusable`. The API key is for private locations, Elasticsearch and Kibana are self-hosted and behind a VPN but I'm connected to the VPN.

Elasticsearch version: 8.11.3
Kibana version: 8.11.3
Synthetics version: 1.9.0
Node version: 21.7.3

### PR DESCRIPTION
**Pull request by vigneshshanmugam**
+ fix #920 
+ Underlying error was triggered from Undici HTTP client when we consume the Response body twice during the error scenario, This hides the error message and throws the default TypeError instead. Copying docs from undici
```
Note: Once a mixin has been called then the body cannot be reused, thus calling additional mixins on .body, e.g. .body.json(); .body.text() will result in an error TypeError: unusable being thrown and returned through the Promise rejection.


```

### CODE REVIEW

File: src/push/request.ts [hunk 1]
```diff
@@ -80,14 +80,11 @@ export async function handleError(
     throw formatNotFoundError(url, await body.text());
   } else if (!ok(statusCode)) {
     let parsed: APIError;
+    const resp = await body.text();
     try {
```
Comment (emilioalvap) [inline src/push/request.ts:84]:
```suggestion

    const resp = await body.text();

```

Comment (emilioalvap) [reply src/push/request.ts:84]: We should guard against `body.text()` rejections

Comment (vigneshshanmugam) [reply src/push/request.ts:84]: Good catch. I will move the error message as generic as well.

**Review by emilioalvap — APPROVED** [review]
LGTM

**Pull request merged**